shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.mm

```diff
@@ -0,0 +1,169 @@
+                                                      inLanguage:language
+                                                  startingOffset:nextOffset];
+    if (nextSpellSuggestion != nil) {
+      [allSpellSuggestions addObject:nextSpellSuggestion];
+      nextOffset =
+          nextSpellSuggestion.misspelledRange.location + nextSpellSuggestion.misspelledRange.length;
+    }
+  } while (nextSpellSuggestion != nil && nextOffset < text.length);
+
+  NSMutableArray* methodChannelResult = [[[NSMutableArray alloc] init] autorelease];
+
+  for (FlutterSpellCheckResult* result in allSpellSuggestions) {
+    [methodChannelResult addObject:[result toFormattedString]];
+  }
+
+  [allSpellSuggestions release];
+  return methodChannelResult;
+}
+
+// Get the misspelled word and suggestions.
+//
+// Returns nil if no spell check suggestions.
+- (FlutterSpellCheckResult*)findSpellCheckSuggestionsForText:(NSString*)text
+                                                  inLanguage:(NSString*)language
+                                              startingOffset:(NSInteger)startingOffset {
+  FML_DCHECK([UITextChecker.availableLanguages containsObject:language]);
+  NSRange misspelledRange =
+      [self.textChecker rangeOfMisspelledWordInString:text
+                                                range:NSMakeRange(0, text.length)
+                                           startingAt:startingOffset
+                                                 wrap:NO
+                                             language:language];
+  if (misspelledRange.location == NSNotFound) {
+    // No misspelled word found
+    return nil;
+  }
+
+  // If no possible guesses, the API returns an empty array:
+  // https://developer.apple.com/documentation/uikit/uitextchecker/1621037-guessesforwordrange?language=objc
+  NSArray<NSString*>* suggestions = [self.textChecker guessesForWordRange:misspelledRange
+                                                                 inString:text
+                                                                 language:language];
+  FlutterSpellCheckResult* result =
+      [[[FlutterSpellCheckResult alloc] initWithMisspelledRange:misspelledRange
+                                                    suggestions:suggestions] autorelease];
+  return result;
+}
+
+- (UITextChecker*)textChecker {
+  return _textChecker;
+}
+
+- (void)dealloc {
+  [_textChecker release];
+  [super dealloc];
+}
+
+@end
+
+@implementation FlutterSpellCheckResult
+
+- (instancetype)initWithMisspelledRange:(NSRange)range
+                            suggestions:(NSArray<NSString*>*)suggestions {
+  self = [super init];
+  if (self) {
+    _suggestions = [suggestions copy];
+    _misspelledRange = range;
+  }
+  return self;
+}
+
+- (NSString*)toFormattedString {
+  return FormatSpellCheckResult(_misspelledRange.location,
+                                _misspelledRange.location + _misspelledRange.length - 1,
+                                [_suggestions copy]);
+}
```

Comment:
This is the 2nd different between the original PR. Instead of formatting the result as a dictionary in my original PR: https://github.com/flutter/engine/pull/32941/files#diff-1c2372162cba7041e14181d7d080cbb0c66eeb44e9cd75379ab8969646e3087bR154-R159

I formatted as a String that matches the android format: https://github.com/flutter/engine/blob/main/shell/platform/android/io/flutter/plugin/editing/SpellCheckPlugin.java#L108-L110
Note that there is a typo in the android documentation, it is suppose to be \n not /n